src/librustc_mir/borrow_check.rs

```diff
@@ -1234,6 +1237,18 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
                         ("",   format!(""), None), // (dont emit downcast info)
                     ProjectionElem::Field(field, _ty) => {
                         autoderef = true;
+                        match proj.base {
+                            Lvalue::Local(local) => {
+                                let local = &self.mir.local_decls[local];
+                                match local.ty.sty {
+                                    ty::TyClosure(_, _) => {
+                                        is_projection_from_ty_closure = true;
+                                    },
+                                    _ => ()
+                                }
+                            },
+                            _ => ()
+                        };
                         ("", format!(".{}", self.describe_field(&proj.base, field.index())), None)
```

Comment:
this looks like it would print `.local`. I think it's probably a better idea to ditch the prefix/middle/suffix scheme and just push things from the match arms.